src/checkers/lines.rs

```diff
@@ -60,6 +67,27 @@ pub fn check_lines(
                     diagnostics.push(diagnostic);
                 }
             }
+
+            if enforce_shebang_whitespace || enforce_shebang_newline || enforce_shebang_python {
```

Comment:
@sbrugman - I did some benchmarking and using a single `extract_shebang` call (+ the weird optimization track I added to the extraction method) brought time down from ~180ms to 100ms.

Reply:
(Just noting here so that you're not confused if you come back to finish off the plugin.)

Reply:
Excellent, thanks. I considered it but was not sure if having the logic in `lines.rs` was ok, or that is should have been moved to the `flake8-executable` rules.